scraper/src/scraper/providers/moonshot.py

```
@@ -0,0 +1,90 @@
+
+    fallback_map: dict[str, tuple[float, float]] = {}
+    if existing:
+        for m in existing:
+            if m.provider == _PROVIDER:
+                fallback_map[m.name] = (m.price_in, m.price_out)
+    for k, v in _FALLBACKS.items():
+        fallback_map.setdefault(k, v)
+
+    try:
+        html = get_page_text(_URL, timeout_ms=40_000)
+    except Exception as exc:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)